File: C  programs/Problems on Pattern Printing/Rows & Columns/Pattern6.c
/* Write a program which accept the number of rows and number of columns from user and Display below pattern.

Input:  iRow=4  iCol=4
Output : 
        # # # #
        * * * *
        # # # #
        * * * *
*/
		  
#include<stdio.h>

void Display(int iRow,int iCol)
{
	int i=0;
	int j=0;
    
    if(iRow<0)
	{
	   iRow=-iRow;
	}

	if(iCol<0)
	{
	   iCol=-iCol;
	}

	//    1    2     3
	for(i=1;i<=iRow;i++)  //Outer
	{
		//   1     2     3
	   for(j=1;j<=iCol;j++)  //Inner 
	   {
		  if((i%2)==0)
		  {
		     printf("*\t");   //Even
		  }
          else
          {
			 printf("#\t");  //Odd
	      } 
	   }
	   printf("\n");
	}   
}

int main()
{
  int iValue1=0,iValue2=0;

  printf("Enter number of rows\n");
  scanf("%d",&iValue1);

  printf("Enter number of columns\n");
  scanf("%d",&iValue2);

  Display(iValue1,iValue2); 

  return 0;
}  	  

File: Java programs/Problems on Pattern Printing/Print square and number pattern/Pattern6.java
/* Write a java program which accept number of rows and number of columns from user and display below pattern.

Input : iRow = 4 iCol = 4
Output :    1 2 3 4 5
            1 2     5
            1   3   5
            1     4 5
            1 2 3 4 5       

*/  

import java.util.*;

class Pattern
{
   void Pattern(int iRow,int iCol)
   {
	   int i=0;
	   int j=0;
	   for(i=1;i<=iRow;i++)        
	   {
		 for(j=1;j<=iCol;j++)
		 {
			if ((j==1)||(i==1)||(j==iCol)||(i==iRow))
			{
			   System.out.print(j+"\t");
			}
			else if(i==j)
			{
			   System.out.print(i+"\t");
			}
			else
			{
			   System.out.print("\t");
			}
		  }	
		  System.out.println();
        } 
    }
}

class Lba405
{
	public static void main(String arg[])
	{
		int iValue1=0;
		int iValue2=0;
		
		Scanner sobj=new Scanner(System.in);
		
		System.out.println("Enter number of rows");
		iValue1=sobj.nextInt();
		
		System.out.println("Enter number of columns");
		iValue2=sobj.nextInt();
		
		Pattern pobj=new Pattern();
		pobj.Pattern(iValue1,iValue2);
	}	
}

